runtime/lua/vim/treesitter/_query_linter.lua

```diff
@@ -57,21 +65,23 @@ local function guess_query_lang(buf)
   end
 end
 
----@private
----@param buf (integer)
----@param opts (QueryLinterOpts|QueryLinterNormalizedOpts|nil)
----@return (QueryLinterNormalizedOpts)
+--- @private
+--- @param buf integer
+--- @param opts QueryLinterOpts|QueryLinterNormalizedOpts|nil
+--- @return QueryLinterNormalizedOpts
 local function normalize_opts(buf, opts)
   opts = opts or {}
   if not opts.langs then
     opts.langs = guess_query_lang(buf)
   end
+
   if type(opts.langs) ~= 'table' then
+    --- @diagnostic disable-next-line:assign-type-mismatch
     opts.langs = { opts.langs }
   end
 
-  ---@cast opts QueryLinterNormalizedOpts
-  opts.langs = opts.langs or { nil }
```

Comment:
this needs to be `{ nil }` be cause we want one iteration with `lang = nil`

Reply:
Ooops, need to fix that. Forgot what was necessary to have nil entries in list-like tables. Will replace that with explict logic.

Reply:
You can't iterate a table with nil in

Reply:
replaced the loop with `for i = 1, math.max(1, #opts.langs)